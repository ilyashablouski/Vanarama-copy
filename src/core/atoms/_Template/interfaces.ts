import { IBaseProps } from '../../interfaces/base';

export interface ITemplateProps extends IBaseProps {}
