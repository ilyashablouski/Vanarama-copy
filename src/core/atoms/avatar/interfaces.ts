import { IBaseProps } from '../../interfaces/base';
import { TSize } from '../../../types/size';

export interface IAvatarProps extends IBaseProps {
  src?: string;
  size?: TSize;
  altText?: string;
}
