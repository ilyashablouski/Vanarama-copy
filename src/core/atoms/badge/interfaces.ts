import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';
import { TSize } from '../../../types/size';

export interface IBadgeProps extends IBaseProps {
  size?: TSize;
  color?: TColor;
  label?: string;
}
