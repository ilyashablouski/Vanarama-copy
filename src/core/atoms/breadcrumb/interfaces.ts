import { IBaseProps } from '../../interfaces/base';
import { ILink } from '../../interfaces/link';

export interface IBreadcrumbProps extends IBaseProps {
  items?: ILink[];
}
