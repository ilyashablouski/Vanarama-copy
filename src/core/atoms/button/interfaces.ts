import { ButtonHTMLAttributes } from 'react';
import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';
import { TSize } from '../../../types/size';

export interface IButtonProps
  extends IBaseProps,
    ButtonHTMLAttributes<HTMLButtonElement> {
  color?: TColor;
  fill?: 'solid' | 'outline' | 'clear';
  icon?: React.ReactNode;
  iconColor?: TColor;
  iconPosition?: 'before' | 'after';
  label?: string | React.ReactNode;
  round?: boolean;
  size?: TSize;
  withoutDefaultClass?: boolean;
  customCTAColor?: string;
}
