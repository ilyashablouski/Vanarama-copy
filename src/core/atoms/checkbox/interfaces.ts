import { InputHTMLAttributes, ReactElement } from 'react';
import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';

export interface ICheckboxProps
  extends IBaseProps,
    InputHTMLAttributes<HTMLInputElement> {
  /**
   * A color for the checkbox.
   */
  color?: TColor;
  /**
   * A unique ID for the checkbox node - this is required here because the
   * underlying checkbox is hidden and therefore you can only click on the label
   * and the label needs to know which input it corresponds to.
   */
  id: string;
  /**
   * 	Invalid state of the checkbox.
   */
  invalid?: boolean;
  /**
   * A name for the input node.
   */
  label: string | (string | ReactElement)[];
  /**
   * Outlined state of the checkbox.
   */
  outline?: boolean;
  /**
   * A ref to attach to the checkbox.
   */
  ref?: React.Ref<HTMLInputElement>;
}
