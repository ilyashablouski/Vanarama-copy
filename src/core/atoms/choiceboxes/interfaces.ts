import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';

export interface IChoice {
  label: string;
  active: boolean;
  value?: string;
}

export interface IChoiceboxesProps extends IBaseProps {
  color?: TColor;
  choices: IChoice[];
  onSubmit: (choice: IChoice) => void;
  multiSelect?: boolean;
  disabled?: boolean | undefined;
  clearMultiSelectTitle?: string;
  onClearClick?: () => void;
  withIcons?: boolean;
  currentValue?: string | string[];
  shouldSelectTheOnlyValue?: boolean;
  setIndex?: number;
}
