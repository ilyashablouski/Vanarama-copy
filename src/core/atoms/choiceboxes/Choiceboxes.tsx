/* eslint-disable @typescript-eslint/no-unused-expressions */
import React, {
  useState,
  forwardRef,
  useImperativeHandle,
  useEffect,
} from 'react';
import cx from 'classnames';
import { IChoiceBoxesProps, IChoice } from './interfaces';
import Icon from '../icon';
import BodyStyleIconMap from '../../../utils/bodyStyleIconMap';
import { getPartnerProperties } from '../../../utils/partnerProperties';

const Choiceboxes = forwardRef(
  (
    {
      className,
      color = 'teal',
      choices,
      multiSelect,
      onSubmit,
      disabled,
      clearMultiSelectTitle,
      onClearClick,
      withIcons,
      currentValue,
      shouldSelectTheOnlyValue = false,
      choiceIndex,
      setChoiceIndex,
      boxClassName,
      labelClassName,
      preventUnselectAllValues = false,
      dataUiTestId,
    }: IChoiceBoxesProps,
    ref,
  ) => {
    const [currentChoices, setCurrentChoices] = useState(choices);
    const [clearMultiSelectActive, setClearMultiSelectActive] = useState(false);
    const [partnershipColor, setPartnerShipColor] = useState<
      string | undefined
    >(undefined);

    const changeChoices = (index: number) => {
      const changedChoices = currentChoices.map(
        (choice: IChoice, number: number) => {
          if (multiSelect) {
            if (index === -1) {
              return {
                ...choice,
                active: false,
              };
            }
            return {
              ...choice,
              active: number === index ? !choice.active : choice.active,
            };
          }
          return {
            ...choice,
            active: number === index,
          };
        },
      );
      const numberOfActiveChoices = changedChoices.reduce(
        (count, choice) => (choice.active ? count + 1 : count),
        0,
      );
      if (preventUnselectAllValues && numberOfActiveChoices === 0) {
        return;
      }
      index !== -1 && onSubmit(changedChoices[index]);
      if (index !== -1 && setChoiceIndex) {
        setChoiceIndex(index);
      }
      setCurrentChoices(changedChoices);
    };

    useEffect(() => {
      if (choiceIndex || choiceIndex === 0) {
        changeChoices(choiceIndex);
      }
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [choiceIndex]);

    useEffect(() => {
      const partner = getPartnerProperties();
      if (partner?.color) {
        setPartnerShipColor(partner.color);
      }
    }, []);

    useImperativeHandle(ref, () => ({
      updateState() {
        setCurrentChoices(choices);
      },
    }));

    useEffect(() => {
      if (currentValue?.length && currentValue[0] === '') {
        setClearMultiSelectActive(true);
      }
      if (currentChoices.length === 1 && shouldSelectTheOnlyValue) {
        changeChoices(0);
      }
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    return (
      <div className={cx('choice-boxes', className, `-${color}`)}>
        {currentChoices.map((choice: IChoice, index: number) => (
          <button
            style={
              choice.active && partnershipColor
                ? {
                    background: `${partnershipColor}`,
                    border: `${partnershipColor}`,
                  }
                : {}
            }
            disabled={disabled}
            key={choice.value}
            type="button"
            className={cx(
              'choice-box',
              boxClassName,
              choice.active ? '-active' : null,
            )}
            onClick={() => {
              changeChoices(index);
              clearMultiSelectActive && setClearMultiSelectActive(false);
            }}
          >
            <span
              className={cx('choice-label', labelClassName)}
              data-uitestid={
                dataUiTestId
                  ? `${dataUiTestId}_span_${choice.label}`
                  : undefined
              }
            >
              {choice.label}
            </span>
            {withIcons && (
              <Icon
                color="teal"
                icon={BodyStyleIconMap.get(choice?.label.replace(/\s+/g, ''))}
                size="xlarge"
                className="md hydrated"
              />
            )}
          </button>
        ))}
        {clearMultiSelectTitle && choices.length !== 1 && (
          <button
            key={clearMultiSelectTitle}
            type="button"
            className={cx('choice-box', { '-active': clearMultiSelectActive })}
            onClick={() => {
              setClearMultiSelectActive(!clearMultiSelectActive);
              changeChoices(-1);
              onClearClick && onClearClick();
            }}
          >
            {clearMultiSelectTitle}
          </button>
        )}
      </div>
    );
  },
);

export default React.memo(Choiceboxes);
