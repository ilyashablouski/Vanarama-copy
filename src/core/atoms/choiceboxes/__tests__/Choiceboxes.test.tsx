import React from 'react';
import renderer from 'react-test-renderer';

import Choiceboxes from '../Choiceboxes';
import { IChoiceBoxesProps } from '../interfaces';

function getComponent(props: IChoiceBoxesProps) {
  return renderer.create(<Choiceboxes {...props} />).toJSON();
}

describe('<Choiceboxes />', () => {
  it('renders correctly', () => {
    const choices = [
      { label: 'Personal', value: 'PCH', active: true },
      { label: 'Business', value: 'BCH', active: false },
    ];

    const tree = getComponent({ choices, onSubmit: () => {} });
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const choices = [
      { label: '13', value: '13', active: true },
      { label: '23', value: '23', active: false },
      { label: '43', value: '43', active: false },
      { label: '45', value: '45', active: false },
      { label: '25763', value: '25763', active: false },
    ];

    const tree = getComponent({ choices, onSubmit: () => {} });
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const choices = [
      { label: '13', value: '13', active: true },
      { label: '23', value: '23', active: false },
      { label: '45', value: '45', active: false },
      { label: '47', value: '47', active: false },
      { label: '25763', value: '25763', active: false },
    ];

    const tree = getComponent({
      choices,
      onSubmit: () => {},
      color: 'black',
      clearMultiSelectTitle: 'clear',
    });
    expect(tree).toMatchSnapshot();
  });
});
