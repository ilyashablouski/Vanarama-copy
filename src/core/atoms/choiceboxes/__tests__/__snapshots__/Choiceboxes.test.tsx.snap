// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Choiceboxes /> renders correctly 1`] = `
<div
  className="choice-boxes -teal"
>
  <button
    className="choice-box -active"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    Personal
  </button>
  <button
    className="choice-box"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    Business
  </button>
</div>
`;

exports[`<Choiceboxes /> renders correctly 2`] = `
<div
  className="choice-boxes -teal"
>
  <button
    className="choice-box -active"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    13
  </button>
  <button
    className="choice-box"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    23
  </button>
  <button
    className="choice-box"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    43
  </button>
  <button
    className="choice-box"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    45
  </button>
  <button
    className="choice-box"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    25763
  </button>
</div>
`;

exports[`<Choiceboxes /> renders correctly 3`] = `
<div
  className="choice-boxes -black"
>
  <button
    className="choice-box -active"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    13
  </button>
  <button
    className="choice-box"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    23
  </button>
  <button
    className="choice-box"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    45
  </button>
  <button
    className="choice-box"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    47
  </button>
  <button
    className="choice-box"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    25763
  </button>
  <button
    className="choice-box"
    onClick={[Function]}
    type="button"
  >
    clear
  </button>
</div>
`;
